src/api.js

```diff
@@ -789,7 +789,13 @@ API.prototype.declare = function(options, handler) {
   if (this._entries.filter(entry => entry.route == options.route && entry.method == options.method).length > 0) {
     throw new Error('Identical route and method declaration.');
   }
-  this._entries.push(options);
+  if (this._entries.map(a => a.name).some(function(element) {
+    return element == options.name;
+  })) {
```

Comment:
In ES6, you can use arrow functions instead of `function`, and they fit on a single line a lot better.  Also, there's no need to do the mapping separately from the check.  How about
`if (this._entries.some(entry => entry.name === options.name))`